crates/services/gas_price_service/src/fuel_gas_price_updater.rs

```diff
@@ -24,12 +24,9 @@ pub mod fuel_core_storage_adapter;
 
 pub mod da_source_adapter;
 
-pub struct FuelGasPriceUpdater<L2, Metadata, DaBlockCosts> {
+pub struct FuelGasPriceUpdater<Metadata> {
```

Comment:
Hmm. I'm not sure how much we are actually gaining from this. I was kinda thinking that we could get rid of the `UpdateAlgorithm` abstraction completely, including removing this struct.

But I don't think the `gas-price-algorithm` should depend directly on some algorithm, in which case we want to keep the `UpdateAlgorithm` trait.

This approach commits us to a specific trait interface with
```rs
        l2_block: BlockInfo,
        da_block_costs: Option<DaBlockCosts>,
```
whereas before the `gas-price-service` didn't know anything about the dependencies of the updater.

Reply:
We are going to want to add another value here soon with this issue:
https://github.com/FuelLabs/fuel-core/issues/2166

Reply:
okay. then I would propose to move the L2 source and DA back into the `FuelGasPriceUpdater` and delegate sub-service startup and shutdown to it. does that sound okay?

Reply:
(it also means we turn `FuelGasPriceUpdater` into a RunnableTask)